File: Naive_String_Matching/Naive_Approach.java
class Main
{
    public static void search(String text, String pattern)
    {
        int lengthText = text.length();
        int lengthPattern = pattern.length();

        for(int i = 0; i <= lengthText - lengthPattern; i++)
        {
            int j;

            for(j = 0; j < lengthPattern; j++)
                if(text.charAt(i + j) != pattern.charAt(j))
                    break;

            if(j == lengthPattern)
               System.out.println("Pattern found at " + (i + 1));
        }
    }

    public static void main(String[] args)
    {
       String text = "namanchamanbomanamansanam";
       String pattern = "aman";
       search(text, pattern);
    }
}

/* Output

Pattern found at 2
Pattern found at 8
Pattern found at 17

*/

File: Naive_String_Matching/Naive_Approach.rb
def search (text, pattern)
  for i in (0 .. text.length - pattern.length)
    j = 0
    while j < pattern.length
      if text[i+j] != pattern[j]
        break
      end
        j += 1
    end
    if j == pattern.length
      puts("Pattern found at " + (i+1).to_s)
    end
  end
end

text = "random variables are so random"
pattern = "ran"
search(text,pattern)

# Output
# Pattern found at 1
# Pattern found at 25
